server/src/main/scala/org/http4s/server/logging/ServerLogging.scala

```diff
@@ -0,0 +1,46 @@
+package org.http4s
+package server
+package logging
+
+import java.time.format.DateTimeFormatter
+import java.time.{Clock, OffsetDateTime}
+
+import org.http4s.headers.Authorization
+
+abstract class ServerLogging[F[_]] {
+  def enabled: Boolean
+  def logRequestResponse(request: Request[F], response: Response[F]): Unit
+}
+
+object ServerLogging {
+  def apply[F[_]](log: (Request[F], Response[F]) => Unit): ServerLogging[F] =
+    new ServerLogging[F] {
+      override final val enabled = true
+      override def logRequestResponse(request: Request[F], response: Response[F]): Unit =
+        log(request, response)
+    }
+
+  def disabled[F[_]]: ServerLogging[F] = new ServerLogging[F] {
+    override final val enabled: Boolean = false
+    override def logRequestResponse(request: Request[F], response: Response[F]): Unit = ()
+  }
+
+  def commonLogFormat[F[_]](clock: Clock = Clock.systemDefaultZone()): ServerLogging[F] = {
+    val formatter = DateTimeFormatter.ofPattern("dd/MMM/yyyy:hh:mm:ss Z")
+
+    apply[F] { (req, res) =>
+      val remotehost = req.remoteAddr.getOrElse("-")
+      val rfc931 = "-"
+      val authuser = Authorization.from(req.headers).map(_.credentials) match {
+        case Some(BasicCredentials(credentials)) => credentials.username
+        case _ => "-"
+      }
+      val date = formatter.format(OffsetDateTime.now(clock))
+      val request = s"${req.method} ${req.uri} ${req.httpVersion}"
+      val status = res.status.code
+      val bytes = res.contentLength.map(_.toString).getOrElse("-")
+
+      println(s"""$remotehost $rfc931 $authuser [$date] "$request" $status $bytes""")
```

Comment:
We might definitely want a custom logger here, but definitely not println.

That aside, I kid you not if we're looking @ performance heavy code, we're better off hand-rolling string builders.

Reply:
Should probably pass this a logger or a `String => ()`.

There's a [fast string interpolator macro](https://github.com/sizmek/fast-string-interpolator), but I don't want to add a dependency for it.  I hate to make code crappier without a benchmark, but the `StringBuilder` seems like a good way out of if we can demonstrate a problem.

Reply:
An instance of ServerLogging can be implemented to use a logger. The reason the commonLogFormat-instance is using println is exactly to avoid having a logger attach a bunch of timestamps and other crap that would make it output something other than Common Log Format.

Reply:
I agree on trying to not output cruft, but I think the logger can be a custom one so it doesn't go to stdout.